client/lib/catch-js-errors/index.js

```diff
@@ -11,24 +11,27 @@ export default class ErrorLogger {
 
 		if ( ! window.onerror ) {
 			TraceKit.report.subscribe( errorReport => {
+				const error = {
+					message: errorReport.message,
+					url: document.location.href
+				};
+
 				if ( Array.isArray( errorReport.stack ) ) {
-					errorReport.stack.forEach( report => Object.keys( report ).forEach( key => {
+					const trace = errorReport.stack.slice( 0, 10 );
+					trace.forEach( report => Object.keys( report ).forEach( key => {
 						if ( key === 'context' && report[ key ] ) {
-							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 512 );
+							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 256 );
```

Comment:
That is only for `key === 'context'`because that field does not need any more data  and is always huge.
Limit of 512 is for all other `report[ key ]` strings. Since these strings are almost always much much smaller, we can allow limit to be higher to only truncate very rare exceptions.